Add Tweet Tokenizer (#13)
JuliaText/WordTokenizers.jl: README.md

`split(foo, Words)` is the same as `tokenize(foo)`,  
and  
`split(foo, Sentences)` is the same as `split_sentences(foo)`.

## Using TokenBuffer API for Custom Tokenizers
We offer a `TokenBuffer` API and supporting utility parsers
for high speed tokenization.

> #### Writing your own TokenBuffer parsers

Reviewer: ```suggestion
#### Writing your own TokenBuffer tokenizer
```

`TokenBuffer` turns a string into a readable stream, used for building tokenizers.

> Utility parsers such as `spaces` and `number` read characters from the

Reviewer: ```suggestion
Utility lexers such as `spaces` and `number` read characters from the
```

stream and into an array of tokens.

> Parsers return `true` or `false` to indicate whether they matched

Reviewer: ```suggestion
Lexers return `true` or `false` to indicate whether they matched
```

        return ts.tokens
    end

    tokenise("foo bar baz") # ["foo", "bar", "baz"]

Many prewritten components for building custom tokenizers
can be found in `src/words/fast.jl` and `src/words/tweet_tokenizer.jl`
These components can be mixed and matched to create more complex tokenizers.

Here is a more complex example.

```julia
julia> using WordTokenizers: TokenBuffer, isdone, character, spaces # Present in fast.jl

julia> using WordTokenizers: nltk_url1, nltk_url2, nltk_phonenumbers # Present in tweet_tokenizer.jl

julia> function tokeinze(input)
           urls(ts) = nltk_url1(ts) || nltk_url2(ts)

           ts = TokenBuffer(input)
           while !isdone(ts)
               spaces(ts) && continue
               urls(ts) ||
               nltk_phonenumbers(ts) ||
               character(ts)
           end
           return ts.tokens
       end
tokeinze (generic function with 1 method)

julia> tokeinze("A url https://github.com/JuliaText/WordTokenizers.jl/ and phonenumber +0 (987) - 2344321")
6-element Array{String,1}:
 "A"
 "url"
 "https://github.com/JuliaText/WordTokenizers.jl/" # URL detected.
 "and"
 "phonenumber"
 "+0 (987) - 2344321" # Phone number detected.
```

> #### Tips for writing custom tokenizers and your own TokenBuffer Parser/Feature

Reviewer: ```suggestion
#### Tips for writing custom tokenizers and your own TokenBuffer Lexer
```

> 1. The order in which the parsers are written needs to be taken care of in some cases-

Reviewer: ```suggestion
1. The order in which the lexers are written needs to be taken care of in some cases-
```

and split about it.

```julia
julia> using WordTokenizers: TokenBuffer, isdone, character, spaces, nltk_phonenumbers, number

julia> order1(ts) = number(ts) || nltk_phonenumbers(ts)
order1 (generic function with 1 method)

julia> order2(ts) = nltk_phonenumbers(ts) || number(ts)
order2 (generic function with 1 method)

julia> function tokenize1(input)
           ts = TokenBuffer(input)
           while !isdone(ts)
               order1(ts) ||
               character(ts)
           end
           return ts.tokens
       end
tokenize1 (generic function with 1 method)

julia> function tokenize2(input)
           ts = TokenBuffer(input)
           while !isdone(ts)
               order2(ts) ||
               character(ts)
           end
           return ts.tokens
       end
tokenize2 (generic function with 1 method)

julia> tokenize1("987-654-3210") # number(ts) || nltk_phonenumbers(ts)
5-element Array{String,1}:
 "987"
 "-"
 "654"
 "-"
 "3210"

julia> tokenize2("987-654-3210") # nltk_phonenumbers(ts) || number(ts)
1-element Array{String,1}:
 "987-654-3210"
```

2. BoundsError and errors while handling edge cases are most common

> and need to be taken of while writing the TokenBuffer parsers.

Reviewer: ```suggestion
and need to be taken of while writing the TokenBuffer lexers.
```

 "987"
 "-"
 "654"
 "-"
 "3210"

julia> tokenize2("987-654-3210") # nltk_phonenumbers(ts) || number(ts)
1-element Array{String,1}:
 "987-654-3210"
```

#### Writing your own TokenBuffer parsers

`TokenBuffer` turns a string into a readable stream, used for building tokenizers.
Utility parsers such as `spaces` and `number` read characters from the
stream and into an array of tokens.

Parsers return `true` or `false` to indicate whether they matched
in the input stream. They can therefore be combined easily, e.g.

    spacesornumber(ts) = spaces(ts) || number(ts)

either skips whitespace or parses a number token, if possible.

The simplest possible tokenizer accepts any `character` with no token breaks:

    function tokenise(input)
        ts = TokenBuffer(input)
        while !isdone(ts)
            character(ts)
        end
        return ts.tokens
    end

    tokenise("foo bar baz") # ["foo bar baz"]

The second simplest splits only on spaces:

    function tokenise(input)
        ts = TokenBuffer(input)
        while !isdone(ts)
            spaces(ts) || character(ts)
        end
        return ts.tokens
    end

    tokenise("foo bar baz") # ["foo", "bar", "baz"]

> You may see `nltk_word_tokenize` for a more advanced example.

Reviewer: Mention something here about 
that many prewritten components for building custom tokenizers,
can be found in <link to source code1>, and <link to source code2>, 
and that these can be mixed and matched
to achieve the totkenization rules you desire.

I would also remove the character example and go straight to:
_"The simplest useful tokenizer splits on spaces"_
then follow up with an example that splits 
URLs and phone numbers and spaces.

Author: > Mention something here about
that many prewritten components for building custom tokenizers,
can be found in , and ,

Author: Should I list out the components (parsers) for custom tokenizers?
